duckduckgo/iOS
# Clean up subscriptionFeature on init

Task/Issue URL:  https://app.asana.com/0/1204099484721401/1207276961185256/f

**Description**:
- Fixes an issue causing welcome page buttons not working when visiting from /pro on active subscription my moving subfeature cleanp to initialization.



**Steps to test this PR**:
1.  Get a subscription
2. Remove linked email
3. Go to http://duckduckgo.com/pro
4. Confirm you're taken to the welcome page
5. Add an email
6. Confirm "Done" button takes you back to settings
7. Close settings
8. Go to http://duckduckgo.com/pro
9. Confirm you're taken to the welcome page
10. Tap Any feature and confirm it works.
12. Remove subscription from device
13. Purchase subscription via Apple Store
14. Confirm Welcome page links work there too





---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
